mitosheet/src/widget.tsx

```diff
@@ -201,9 +199,25 @@ export class ExampleView extends DOMWidgetView {
         }
         return userProfile;
     }
-    
+
     getAnalysisData(): AnalysisData {
         const unparsed = this.model.get('analysis_data_json')
-        return JSON.parse(unparsed);
+        const parsed = JSON.parse(unparsed)
+
+        // Parse the nested graphDataJson
+        const graphDataJSON: GraphDataJSON = parsed["graphDataJSON"] as GraphDataJSON
+
+        // Convert empty string to undefined to adhere to best practices
+        Object.entries(graphDataJSON).forEach(([key, val]) => {
+            if (val["graphHTML"] == '') {
+                graphDataJSON[key]["graphHTML"] = undefined
+            }
+            if (val["graphScript"] == '') {
+                graphDataJSON[key]["graphScript"] = undefined
+            }
+        })
+        parsed["graphDataJSON"] = graphDataJSON
```

Comment:
Ah! I see what you're doing here. Gotcha. So ignore my above comment...

This is an interesting example of this application of "encode failure in the type system" - somewhat different than before. A few thoughts:
1. If we're gonna make them undefined, let's do it on the backend, so what we're passing doesn't need to be patched up. It kinda makes it so you need to think about the graph data json being "created" in multiple places, which is bad news. 
2. Also, not doing it on the backend makes it so we have this grow code here vs. it just being a single json.parse, as it was before - which was sweet!
3. Also, since graphHTML and graphScript are both either defined or empty strings together, does it make sense to nest them in a single entity? What about the following:

```
export type GraphData = {
     graphParams: GraphParams,
     graphOutput?: {
          graphGeneratedCode: string,
          graphHTML: string,
          graphScript: string,
     }
 };
```

Note here that: 
1.  The backend would do this (and it would remove the use of the empty string on the backend).
2. The _entire_ output is either there or not. This makes the types more consistent and less likely to be fucky. 

All-in-all, good work thinking about this empty string thing when I did not (per my comment above). It caused me to think about it a bit more as well!